tensorflow_datasets/scripts/replace_fake_images.py

```diff
@@ -83,8 +81,8 @@ def rewrite_zip(root_dir, zip_filepath):
       zip_file.extractall(path=temp_dir)
 
     rewrite_dir(temp_dir)  # Recursivelly compress the archive content
-    shutil.make_archive(zip_filepath[:-4], 'zip', temp_dir) #Copies the directory structure of the extracted file ans stores it back as zip
-    # The extraction target in the above line can be changed to gztar for additional compression 
+    if zip_filepath.lower().endswith('zip'):
+        shutil.make_archive(zip_filepath[:-len('.zip')], 'zip', temp_dir) #Copies the directory structure of the extracted file ans stores it back as zip
```

Comment:
@MadhavEsDios It's look great, after all things please make sure to check code style as described [here](https://www.tensorflow.org/datasets/add_dataset#5_check_your_code_style). 
Also is it solve problem of .tar files like in `Imagenet2012.tar` file, also with remaining issues?

Reply:
Thanks for pointing out the code style, I have fixed it.
No, it does not solve the problem of .tar files (at least for Imagenet2012). As I have written in a previous issue before, this does not seem to be a python tarfile problem, but more of a 'tar' compression format issue. This is because the 'tar' format usually introduces some overhead for hashing internal files.
As mentioned above, I am not entirely sure whether this is the cause, so will still investigate it.
This script fixes .zip file issues, image compression, and absolute path issues along with support for .tgz and .xz files.

Reply:
Thanks for your quick response, please run lint.sh in oss_script as descibed in link i given above as I seen you not run it properly like one  line in code should not be more than 80 words. Can you please describe how use of shutil is advantageous than other low level zip and tar as low level also works  ? What is problem associated with absolute path as i not understand it ?

Reply:
1. Running lint, hi I ran the lint command on the only file I made changes to i.e. replace_fake_images.
`./oss_scripts/lint.sh tensorflow_datasets/scripts/replace_fake_images.py 

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)`
As you can see it gives a score of 10. I have not run it on the whole directory, as I do not want to affect other scripts.So please verify if the warnings that you see are from this particular script. Thanks !

2. Advantages of shutil: I have also mentioned this as a comment below the issue.
If you ran the previous script, the generated zip files contained an unnecessary directory structure like follows:
Let's say the original file was abc.zip and the temp file was temp1, the previous output instead of being abc.zip was '/Users/Name/...../testing/test_data/fake_examples/temp1/abc.zip' instead of just abc.zip. As this was done recursively, lets say this abc.zip file contained another xyz.zip file, this too would use the absolute path instead of just xyz.zip. This issue increased the compressed file size as it contains unnecessary information.
Shutil.make_archive solves this problem by internally using os.chg_dirs and gives the desired output. So the low-level API in the older script was not working as intended.
However, as I have mentioned in a previous issue, modifications to the low-level API can be made by using os library functions to get the desired output. However, using shutil.make_archive is cleaner and has no major disadvantage.

Reply:
Okay it seems good but adding folders in folders not increasing size, still if get off from this problem using shutil thats good.
Maybe it increases some time complexity ? Than recursive approach i am not sure about that still good work. 
Thanks

Reply:
Adding folders within folders did increase the size I think but only for 'zip' files (For eg. cats_vs_dogs). Which is why I guess shutil resolved this issue.
Also, you have rightly pointed out (i have written this in my comment on the issue as well) that using shutil and PIL optimize=True takes longer to execute than the previous version. This is expected behavior.

Thanks for your contributions.